### PR TITLE
Fix wrong version set in TabletMeta and rowset null pointer

### PR DESCRIPTION
**Pull request by chaoyli**

### CODE REVIEW

File: be/src/olap/rowset/alpha_rowset_writer.cpp [hunk 1]
```diff
@@ -154,7 +154,7 @@ RowsetSharedPtr AlphaRowsetWriter::build() {
         return nullptr;
     }
     for (auto& segment_group : _segment_groups) {
-        RETURN_NOT_OK(segment_group->load());
+        segment_group->load();
         bool checked = segment_group->check();
```
Comment (yiguolei) [inline be/src/olap/rowset/alpha_rowset_writer.cpp]: why not check segment group.load?

**Review by yiguolei — APPROVED** [review]

File: be/src/olap/push_handler.cpp [hunk 2]
```diff
@@ -241,175 +241,174 @@ void PushHandler::_get_tablet_infos(const vector<TabletVars>& tablet_vars,
...
 
 OLAPStatus BinaryFile::init(const char* path) {
-  // open file
-  if (OLAP_SUCCESS != open(path, "rb")) {
-    OLAP_LOG_WARNING("fail to open file. [file='%s']", path);
-    return OLAP_ERR_IO_ERROR;
-  }
+    // open file
+    if (OLAP_SUCCESS != open(path, "rb")) {
+        OLAP_LOG_WARNING("fail to open file. [file='%s']", path);
```
Comment (kangpinghuang) [inline be/src/olap/push_handler.cpp]: LOG(WARNING)

File: be/src/olap/push_handler.cpp [hunk 3]
```diff
@@ -241,175 +241,174 @@ void PushHandler::_get_tablet_infos(const vector<TabletVars>& tablet_vars,
...
 
-  // load header
-  if (OLAP_SUCCESS != _header.unserialize(this)) {
-    OLAP_LOG_WARNING("fail to read file header. [file='%s']", path);
-    close();
-    return OLAP_ERR_PUSH_INIT_ERROR;
-  }
+    // load header
+    if (OLAP_SUCCESS != _header.unserialize(this)) {
+        OLAP_LOG_WARNING("fail to read file header. [file='%s']", path);
```
Comment (kangpinghuang) [inline be/src/olap/push_handler.cpp]: LOG(WARNING)

File: be/src/olap/push_handler.cpp [hunk 4]
```diff
@@ -241,175 +241,174 @@ void PushHandler::_get_tablet_infos(const vector<TabletVars>& tablet_vars,
...
+                res = OLAP_ERR_INPUT_PARAMETER_ERROR;
+                break;
+            }
+
+            // create BinaryReader
+            bool need_decompress = false;
+            if (_request.__isset.need_decompress && _request.need_decompress) {
+                need_decompress = true;
+            }
+            if (NULL == (reader = IBinaryReader::create(need_decompress))) {
```
Comment (kangpinghuang) [inline be/src/olap/push_handler.cpp]: (reader = IBinaryReader::create(need_decompress)) == nullptr

File: be/src/olap/snapshot_manager.cpp [hunk 5]
```diff
@@ -279,7 +279,7 @@ string SnapshotManager::_get_header_full_path(
 
 void SnapshotManager::update_header_file_info(
         const vector<RowsetSharedPtr>& consistent_rowsets,
-        TabletMeta* tablet_meta) {
```
Comment (kangpinghuang) [inline be/src/olap/snapshot_manager.cpp]: why change pointer to shared_ptr?

File: be/src/olap/tablet.cpp [hunk 6]
```diff
@@ -177,19 +173,19 @@ OLAPStatus Tablet::revise_tablet_meta(const TabletMeta& tablet_meta,
     OLAPStatus res = OLAP_SUCCESS;
     do {
         // load new local tablet_meta to operate on
-        TabletMeta new_tablet_meta;
-        TabletMetaManager::get_header(_data_dir, tablet_id(), schema_hash(), &new_tablet_meta);
+        TabletMetaSharedPtr new_tablet_meta(new TabletMeta());
```
Comment (kangpinghuang) [inline be/src/olap/tablet.cpp]: TabletMetaSharedPtr new_tablet_meta(new(std::nothrow) TabletMeta())

File: be/src/olap/tablet.h [hunk 7]
```diff
@@ -214,7 +214,7 @@ class Tablet : public std::enable_shared_from_this<Tablet> {
 
 private:
     TabletState _state;
-    TabletMeta* _tablet_meta;
+    std::shared_ptr<TabletMeta> _tablet_meta;
```
Comment (kangpinghuang) [inline be/src/olap/tablet.h]: TabletMetaSharedPtr

File: be/src/olap/tablet_manager.cpp [hunk 8]
```diff
@@ -686,16 +686,15 @@ TabletSharedPtr TabletManager::find_best_tablet_to_compaction(CompactionType com
 
 OLAPStatus TabletManager::load_tablet_from_meta(DataDir* data_dir, TTabletId tablet_id,
         TSchemaHash schema_hash, const std::string& meta_binary) {
-    std::unique_ptr<TabletMeta> tablet_meta(new TabletMeta());
+    TabletMetaSharedPtr tablet_meta(new TabletMeta());
```
Comment (kangpinghuang) [inline be/src/olap/tablet_manager.cpp]: TabletMetaSharedPtr tablet_meta(new(std::nothrow) TabletMeta());

File: be/src/olap/tablet_meta.cpp [hunk 9]
```diff
@@ -54,11 +54,11 @@ OLAPStatus TabletMeta::create(int64_t table_id, int64_t partition_id,
                               uint64_t shard_id, const TTabletSchema& tablet_schema,
                               uint32_t next_unique_id,
                               const std::unordered_map<uint32_t, uint32_t>& col_ordinal_to_unique_id,
-                              TabletMeta** tablet_meta) {
-    *tablet_meta = new TabletMeta(table_id, partition_id,
-                                  tablet_id, schema_hash,
-                                  shard_id, tablet_schema,
-                                  next_unique_id, col_ordinal_to_unique_id);
+                              TabletMetaSharedPtr* tablet_meta) {
+    tablet_meta->reset(new TabletMeta(table_id, partition_id,
```
Comment (kangpinghuang) [inline be/src/olap/tablet_meta.cpp]: std::nothrow

File: be/src/olap/task/engine_storage_migration_task.cpp [hunk 10]
```diff
@@ -127,8 +127,8 @@ OLAPStatus EngineStorageMigrationTask::_storage_medium_migrate(
         }
 
         // generate new header file from the old
-        TabletMeta* new_tablet_meta = new(std::nothrow) TabletMeta();
-        if (new_tablet_meta == NULL) {
+        TabletMetaSharedPtr new_tablet_meta(new(std::nothrow) TabletMeta());
+        if (new_tablet_meta == nullptr) {
             OLAP_LOG_WARNING("new olap header failed");
```
Comment (kangpinghuang) [inline be/src/olap/task/engine_storage_migration_task.cpp]: LOG(WARNING)

File: gensrc/proto/olap_file.proto [hunk 11]
```diff
@@ -270,7 +270,7 @@ message TabletMetaPB {
     optional TabletSchemaPB schema = 9;
     repeated RowsetMetaPB rs_metas = 10;
     repeated RowsetMetaPB inc_rs_metas = 11;
-    optional AlterTabletPB alter_tablet_task = 12;
```
Comment (kangpinghuang) [inline gensrc/proto/olap_file.proto]: why change this name?

File: be/src/olap/push_handler.cpp [hunk 12]
```diff
@@ -241,175 +241,174 @@ void PushHandler::_get_tablet_infos(const vector<TabletVars>& tablet_vars,
...
-                       << " res=" << res << " read_rows=" << num_rows;
-          break;
-        } else {
-          if (OLAP_SUCCESS != (res = rowset_writer->add_row(&row))) {
-            LOG(WARNING) << "fail to attach row to rowset_writer. "
-                         << " res=" << res
-                         << ", tablet=" << curr_tablet->full_name()
-                         << " read_rows=" << num_rows;
+        // 2. init RowsetBuilder of cur_tablet for current push
+        VLOG(3) << "init RowsetBuilder.";
```
Comment (kangpinghuang) [inline be/src/olap/push_handler.cpp]: RowsetWriter

**Pull request merged**